test/mpi_graph_io_tests.cpp

```diff
@@ -0,0 +1,537 @@
+    // The EOF representation in the file evaluates to 10 interpreted as int. In
+    // many cases it actually is represented as a -1:
+    // https://en.cppreference.com/w/cpp/string/char_traits/eof
+    // The following test fails though:
+    // REQUIRE(eof_marker == std::char_traits<char>::eof());
+
+    // File content:
+    // 0 1 1 1
+    // 2 3 1 3
+    // 1 2 1 2
+    // 3 4 1 4
+    // 3 5 1 6
+    // 3 6 1 7
+
+    size_t idx = 0;
+    REQUIRE(timestamped_edges.size() == edge_count);
+    CHECK(timestamped_edges[idx].edge.source == 0);
+    CHECK(timestamped_edges[idx].edge.target.vertex == 1);
+    CHECK(timestamped_edges[idx].edge.target.weight == 1.f);
+    CHECK(timestamped_edges[idx].timestamp == 1);
+
+    ++idx;
+    CHECK(timestamped_edges[idx].edge.source == 2);
+    CHECK(timestamped_edges[idx].edge.target.vertex == 3);
+    CHECK(timestamped_edges[idx].edge.target.weight == 1.f);
+    CHECK(timestamped_edges[idx].timestamp == 3);
+
+    ++idx;
+    CHECK(timestamped_edges[idx].edge.source == 1);
+    CHECK(timestamped_edges[idx].edge.target.vertex == 2);
+    CHECK(timestamped_edges[idx].edge.target.weight == 1.f);
+    CHECK(timestamped_edges[idx].timestamp == 2);
+
+    ++idx;
+    CHECK(timestamped_edges[idx].edge.source == 1);
+    CHECK(timestamped_edges[idx].edge.target.vertex == 4);
+    CHECK(timestamped_edges[idx].edge.target.weight == 1.f);
+    CHECK(timestamped_edges[idx].timestamp == 8);
+
+    ++idx;
+    CHECK(timestamped_edges[idx].edge.source == 3);
+    CHECK(timestamped_edges[idx].edge.target.vertex == 4);
+    CHECK(timestamped_edges[idx].edge.target.weight == 1.f);
+    CHECK(timestamped_edges[idx].timestamp == 4);
+
+    ++idx;
+    CHECK(timestamped_edges[idx].edge.source == 3);
+    CHECK(timestamped_edges[idx].edge.target.vertex == 5);
+    CHECK(timestamped_edges[idx].edge.target.weight == 1.f);
+    CHECK(timestamped_edges[idx].timestamp == 6);
+
+    ++idx;
+    CHECK(timestamped_edges[idx].edge.source == 3);
+    CHECK(timestamped_edges[idx].edge.target.vertex == 6);
+    CHECK(timestamped_edges[idx].edge.target.weight == 1.f);
+    CHECK(timestamped_edges[idx].timestamp == 7);
+
+    ++idx;
+    REQUIRE(timestamped_edges.size() == idx);
+}
+
+
+TEST_CASE("MPI, read_binary_graph, undirected, unweighted, static")
+{
+    Edges32 edges;
+    auto read_f = [&](MPI_File binary_graph)
+    {
+        edges.push_back(Edge32{});
+
+        MPI_Status status;
+        MPI_File_read(binary_graph, &edges.back().source, 1, MPI_INT32_T, &status);
+        MPI_File_read(binary_graph, &edges.back().target, 1, MPI_INT32_T, &status);
+
+        return true;
+    };
+
+    std::filesystem::path file_path(graph_path + unweighted_directed_graph_enzymes_bin);
+    mpi::FileWrapper binary_graph{ file_path };
+
+    auto const [vertex_count, edge_count] = mpi::read_binary_graph(binary_graph.get(), std::move(read_f));
+
+    // TODO: Currently there MPI_ERRROR returns 2: invalid count argument. Fix that!
```

Comment:
still valid?

Reply:
Yes, but I can't get this to work currently. I elaborated on the issue and will leave it as is for this PR.